### PR TITLE
Fix-81729 Preserve case in Search and Replace isn't working when I use group substitution

### PR DESCRIPTION
**Pull request by skprabhanjan**
@roblourens , This was not touched upon when preserve case was implemented , I have changed it to make it work for group substitution also . The gist of the change is that instead of returning `text.replace(this._regExp, this.pattern);` we return `text.replace(this._regExp, this.buildReplaceString(match, preserveCase));`

**NOTE : I have changed test methods to the standard used in other functions to make it look cleaner as this was getting pretty messy , and existing test methods would cover this fix test cases**

#81729

### CODE REVIEW

**Comment by skprabhanjan** [comment]
@roblourens , Any updates on this ?

File: src/vs/workbench/services/search/common/replace.ts [hunk 1]
```diff
@@ -61,7 +61,7 @@ export class ReplacePattern {
 		if (match) {
 			if (this.hasParameters) {
 				if (match[0] === text) {
-					return text.replace(this._regExp, this.pattern);
+					return text.replace(this._regExp, this.buildReplaceString(match, preserveCase));
```
Comment (roblourens) [inline src/vs/workbench/services/search/common/replace.ts]: Can you add tests for this bug?

Comment (skprabhanjan) [reply src/vs/workbench/services/search/common/replace.ts]: Done

**Review by roblourens — APPROVED** [review]
Thanks!

**Pull request merged**